solutions/Pranjal Rai/src/Assignment 1/CommandLineParser.py

```
@@ -0,0 +1,76 @@
+        results = {}
+        commands_found = []
+        error_message = None
+
+        for arg in argv[1:]:
+            command = None
+            value = None
+            argument = arg.split('=')
+
+            if len(argument) == 2:
+                command, value = argument
+            elif len(argument) == 1:
+                command = argument[0]
+            elif len(argument) > 2:
+                error_message = 'invalid number of arguments passed to ' + argument[0]
+
+            commands_found.append(command)
+
+            if command not in self.COMMANDS: # commands with at least one argument
+                error_message = command + ' is not a recognized command'
+            else:
+                if self.COMMANDS[command].is_local is True:
+                    results[command] = True
+                    continue
+                regular_expression = self.COMMANDS[command].regular_expression
+                if re.fullmatch(regular_expression, value):
+                    results[command] = value
+                else:
+                    error_message = 'invalid argument to ' + command
+            
+            if error_message:
+                break
+
+        if error_message is not None:
+            """some error occurred already"""
+            pass
+        else:
+            for command in commands_found:
+                conflicting_command = self.COMMANDS[command].conflicting_command
+                if conflicting_command is not None and conflicting_command in commands_found:
+                    error_message = 'The ' + command + ' and ' + conflicting_command + ' arguments cannot be used together'
+                required_command = self.COMMANDS[command].required_command
+                if required_command is not None and required_command not in commands_found:
+                    error_message = 'The ' + required_command + ' argument is required, but missing from input'
+
+        final_response = error_message
```

Instead of returning an error message, you should raise an exception, so that the application/testing layer can catch it and do whatever it wants with it (print it and exit, or log it and exit, or use some default fallback behavior).
Parsing a string return value (which is what this method generates) is more inconvenient. The API should be: the method either returns a Dict, or raises an exception.